common/tmx_filetype.go

```diff
@@ -0,0 +1,60 @@
+package common
+
+import (
+	"bytes"
+	"fmt"
+        "engo.io/engo"
+	"io"
+)
+
+// TMXResource contains a level created from a Tile Map XML
+type TMXResource struct {
+	Level *Level
+	url   string
+}
+
+func (r TMXResource) URL() string {
+	return r.url
+}
+
+// tmxLoader is responsible for managing '.tmx' files within 'engo.Files'
+type tmxLoader struct {
+	levels map[string]TMXResource
+}
+
+// Load will load the tmx file and any other image resources that are needed
```

Comment:
> it's the tiled map editor format, no one else uses it.

I think @Newbrict and @TheBerryBeast are right in this case, @paked. It's only to be used with the `tmx` format, those that won't be loading `tmx` files won't be using it anyway. This is something to note within some kind of `demo` (which doesn't exist yet? or is that the broken `tilemap` demo? ) - but doesn't really *have* to be here? (could be here, though)